### PR TITLE
23 informers workload

### PR DESCRIPTION
**Pull request by squeedee**

### CODE REVIEW

File: tests/integration/supplychain/workload_reconciler_test.go [hunk 1]
```diff
@@ -274,4 +278,164 @@ var _ = Describe("WorkloadReconciler", func() {
...
+
+			test = &resources.Test{}
+
+			// FIXME: make this more obvious
+			Eventually(func() ([]v1.Condition, error) {
+				err := c.Get(ctx, client.ObjectKey{Name: "test-resource", Namespace: testNS}, test)
+				return test.Status.Conditions, err
+			}).Should(BeNil())
+
+			_ = c.Get(ctx, client.ObjectKey{Name: "test-resource", Namespace: testNS}, test)
```
Comment (emmjohnson) [inline tests/integration/supplychain/workload_reconciler_test.go]: Remove this

File: tests/kuttl/supplychain/templates-ytt/01-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#metadata:
+#  name: responsible-ops---consume-output-of-resources
+#status:
+#  conditions:
+#    - type: TemplatesReady
+#      status: "True"
+#      reason: Ready
+#    - type: Ready
+#      status: "True"
+#      reason: Ready
```
Comment (martyspiewak) [inline tests/kuttl/supplychain/templates-ytt/01-assert.yaml]: Is there a reason this file is commented out?

File: pkg/registrar/map_functions.go [hunk 3]
```diff
@@ -40,6 +40,53 @@ type Mapper struct {
...
+	if err != nil {
+		mapper.Logger.Error(err, "list ClusterSupplyChains")
+		return nil
+	}
+
+	templateKind := template.GetObjectKind().GroupVersionKind().Kind
+
+	var supplyChains []*v1alpha1.ClusterSupplyChain
+	for scIndex := range list.Items {
+		sc := list.Items[scIndex]
```
Comment (martyspiewak) [inline pkg/registrar/map_functions.go]:
```suggestion
	for _, sc := range list.Items {
```
nitpick- this would be more consistent with what we do in `ClusterDeliveryToDeliverableRequests` and `ClusterSupplyChainToWorkloadRequests` below and is a bit easier to reason about while reading through the code.

File: tests/integration/supplychain/policy.yaml [hunk 4]
```diff
@@ -15,11 +15,13 @@
 apiVersion: audit.k8s.io/v1 # This is required.
 kind: Policy
 # Don't generate audit events for all requests in RequestReceived stage.
```
Comment (martyspiewak) [inline tests/integration/supplychain/policy.yaml]:
```suggestion

```
That comment seems to be irrelevant now as it was on the `omitStages` portion that was removed.

**Review by martyspiewak — CHANGES_REQUESTED** [review]

**Comment by martyspiewak** [comment]
Looks great! Left a few comments with some small changes but other than that lgtm!

File: pkg/registrar/map_functions.go (hunk 3 above)
Comment (emmjohnson) [reply pkg/registrar/map_functions.go]: That actually causes major problems. See https://play.golang.org/p/GzU80LswSEt

File: tests/kuttl/supplychain/templates-ytt/01-assert.yaml (hunk 2 above)
Comment (emmjohnson) [reply tests/kuttl/supplychain/templates-ytt/01-assert.yaml]: no

File: pkg/registrar/map_functions.go (hunk 3 above)
Comment (squeedee) [reply pkg/registrar/map_functions.go]: we also inlined a comment to guide people.

**Review by martyspiewak — APPROVED** [review]
